### PR TITLE
Automatically detect OpenSearch version

### PR DESCRIPTION
**Pull request by pavolloffay**
Updates #3044 

* Automatically set ES version to 7 if OpenSearch 1x is used
* Add OpenSearch to itests

### CODE REVIEW

**Comment by pavolloffay** [comment]
@albertteoh would you like to review?

**Review by yurishkuro — DISMISSED** [review]

**Comment by jpkrohling** [comment]
Are the test failures expected?

File: plugin/storage/integration/elasticsearch_test.go [hunk 1]
```diff
@@ -68,6 +69,12 @@ func (s *ESStorageIntegration) getVersion() (uint, error) {
 	if err != nil {
 		return 0, err
 	}
+	// OpenSearch is based on ES 7.x
+	if strings.Contains(pingResult.TagLine, "OpenSearch") {
```
Comment (pavolloffay) [inline plugin/storage/integration/elasticsearch_test.go:73]: this is unfortunate we should refactor the test to use the pkg/es/config

**Comment by pavolloffay** [comment]
> Are the test failures expected?

no it needed a bit more love. It should be fixed now and it needs re-approval.

**Comment by pavolloffay** [comment]
still failed, I will have a look

**Comment by pavolloffay** [comment]
alright so the rollover script needed the "opensearch" logic as well.

**Review by albertteoh — APPROVED** [review]

**Pull request merged**